Program/src/file_delivery_app.c

```diff
@@ -0,0 +1,127 @@
+/*------------------------------------------------------------------------------
+This file is protected under copyright. If you want to use it,
+please include this text, that is my only stipulation.  
+
+Author: Evan Giese
+------------------------------------------------------------------------------*/
+#include "mib.h"
+#include "port.h"
+#include "file_delivery_app.h"
+#include "tasks.h"
+
+static void create_ssp_server_drivers(FTP *app) {
+
+    if (app->remote_entity.type_of_network == posix_connection_less) {
+        app->server_handle = ssp_thread_create(STACK_ALLOCATION, ssp_connectionless_server_task, app);
+
+    } else if(app->remote_entity.type_of_network == posix_connection) {
+        app->server_handle = ssp_thread_create(STACK_ALLOCATION, ssp_connection_server_task, app);
+
+    } else if (app->remote_entity.type_of_network == csp_connectionless) {
+        app->server_handle = ssp_thread_create(STACK_ALLOCATION, ssp_csp_connectionless_server_task, app);
+
+    } else if (app->remote_entity.type_of_network == csp_connection) {
+        app->server_handle = ssp_thread_create(STACK_ALLOCATION, ssp_csp_connection_server_task, app);
+    }
+}
+
+static void create_ssp_client_drivers(Client *client) {
+    Remote_entity remote_entity = client->remote_entity;
+
+    if (remote_entity.type_of_network == posix_connection_less) {
+        client->client_handle = ssp_thread_create(STACK_ALLOCATION, ssp_connectionless_client_task, client);
+
+    } else if(remote_entity.type_of_network == posix_connection) {
+        client->client_handle = ssp_thread_create(STACK_ALLOCATION, ssp_connection_client_task, client);
+
+    } else if (remote_entity.type_of_network == csp_connectionless) {
+        client->client_handle = ssp_thread_create(STACK_ALLOCATION, ssp_csp_connectionless_client_task, client);
+    }
+    else if (remote_entity.type_of_network == csp_connection) {
+        client->client_handle = ssp_thread_create(STACK_ALLOCATION, ssp_csp_connection_client_task, client);
+    }
+}
+
+FTP *init_ftp(uint32_t my_cfdp_address) {
+
+    
+    Remote_entity remote_entity;
+    int error = get_remote_entity_from_json(&remote_entity, my_cfdp_address);
+    if (error == -1) {
+        ssp_error("couldn't start server\n");
```

Comment:
hmm... this could be a design tradeoff. I'll have to see what reporting is in there, it should report something like "cant read in configuration data." This error is a bit too generic. Thanks

Reply:
just checked back on this function. There is quite a bit of reporting (assuming errorno is set down the line). I like to have static error strings close to the code that failed, so I can easily search through logs.

Reply:
changed the error to "cant read in configuration data" much less generic